cmd/frontend/internal/frontend/service.go

```diff
@@ -330,6 +333,12 @@ func (fes *FrontEndService) Monitor(ctx context.Context, errCh chan<- error) {
 				if refreshCancel != nil {
 					refreshCancel()
 				}
+				if hasConnectivity {
+					// (logging will not work inside denounceFrontend())
+					if err := denounceFrontend(context.Background(), fes.targetRegistryClient); err != nil {
```

Comment:
I wonder if blocking the shutdown for 5 seconds could be a bit too much if e.g. NSP is not around.
Anyways, we can revisit it, if required...